.github/workflows/pipeline.yml

```diff
@@ -1,51 +1,44 @@
-name: Linting and tests
+name: CI
 on:
   pull_request:
   push:
     branches: ["main"]
+  workflow_dispatch:
+
 jobs:
-  check-code-style:
+  check-newsfile:
+    name: Check PR has a changelog
+    if: ${{ github.base_ref == 'main' || contains(github.base_ref, 'release-') }}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{github.event.pull_request.head.sha}}
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.6'
-      - run: python -m pip install -e .[dev]
-      # Please keep this section up to date with `scripts-dev/lint.sh`.
-      # TODO: could just run the lint script directly?
-      - run: black --check --diff sydent/ stubs/ tests/ matrix_is_test/ scripts/ setup.py
-      - run: flake8 sydent/ tests/ matrix_is_test/ scripts/ setup.py
-      - run: flake8 stubs/ --ignore E301,E302,E305,E701,E704
-      - run: isort --check-only --diff sydent/ stubs/ tests/ matrix_is_test/ scripts/ setup.py
+          python-version: "3.7"
+      - run: python -m pip install towncrier
+      - run: "scripts-dev/check_newsfragment.sh ${{ github.event.number }}"
+
+  common-python-ci:
+    uses: "matrix-org/backend-meta/.github/workflows/python-poetry-ci.yml@v1"
```

Comment:
What are the semantics of `@v1` here — is that fixed to a literal git tag, or something smarter?
If so, to update it, do we have to come back here and replace it with `@v2`?

Reply:
> What are the semantics of `@v1` here — is that fixed to a literal git tag, or something smarter?

Literal git tag, branch name or SHA hash.

> If so, to update it, do we have to come back here and replace it with `@v2`?

GHA's advice on versioning is [here](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-release-management-for-actions). (That's for a custom action rather than a reusable workflow, but I think the advice is worth following here too.) TL;DR:

- use semver
- tag each released version as vMAJOR.MINOR.PATCH.
- maintain a major version tag vMAJOR.
- if you make a MINOR or PATCH release, move the major version tag to point to the latest release.

Reply:
> if you make a MINOR or PATCH release, move the major version tag to point to the latest release.

Aha, I see — I seem to remember it being bad karma to change a tag but I suppose if it's for an action where this is the guideline, it's fine, I think.